app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -26,11 +24,45 @@ class MainActivity : AppCompatActivity() {
         drawingFieldView.onBitmapUpdate = {
             model.activeBitmap = it
         }
-
+        //when we click to create a new layer
         findViewById<TextView>(R.id.newLayer).setOnClickListener {
-            showLayerAlertDialog()
+            //assign the alert dialog to be able to dismiss later
+            //show the dialog for new layer
+            alertDialog = DialogUtility.showLayerAlertDialog(
+                context = this,
+                layerNames = model.layerNames,
+                wasOpen = model.isLayerCreationDialogOpen,
+                onTextChanged = { model.currentLayerDialogText = it },
+                onDialogSubmission = {
+                    Log.v("Hamdan", "${it}")
+                    if (it != null) {
+                        model.layerNames.add(it)
+                    }
+                    model.isLayerCreationDialogOpen = false
+                },
+                onOpen = { model.isLayerCreationDialogOpen = true }
```

Comment:
is model the reference to the DialogUtility file? or is it the viewModel? or the Adapter LayerViewModel?

my brain is telling me its likely the third one in the list, but this seems like a lot to take in here

Reply:
nope its the viewmodel because were in the activity, the adapter is only involved when using the adapter.